src/index.ts

```diff
@@ -25,16 +27,15 @@ export default async function createAuth0Client(options: Auth0ClientOptions) {
 
   const auth0 = new Auth0Client(options);
 
-  if (!ClientStorage.get('auth0.is.authenticated')) {
+  if (
+    auth0.cacheLocation === CACHE_LOCATION_MEMORY &&
+    !ClientStorage.get('auth0.is.authenticated')
+  ) {
     return auth0;
   }
 
   try {
-    await auth0.getTokenSilently({
-      audience: options.audience,
-      scope: options.scope,
-      ignoreCache: true
-    });
+    await auth0.getTokenSilently();
```

Comment:
I removed these options as `getTokenSilently` will default to the global options that are specified to `createAuth0Client` anyway.

Reply:
It seems that `ignoreCache` is being hardcoded here to `true`, and [if I followed the code correctly](https://github.com/auth0/auth0-spa-js/blob/fb984a66ac72bf2914b7a853fd72bf5678ea2b01/src/Auth0Client.ts#L344), the default value for it is `false`. That would be a breaking change in behavior. If that's true, then I think we should keep that option at least here.

Reply:
Yes you followed correctly. The actual intention here was that actually we could use the cache on startup if we have one, considering that the SDK now has persistence.

I'm not sure this would be a breaking change:

* If you're using the in-memory cache, there won't be anything in the cache anyway, making the value here redundant
* If you are using local storage, we might have something in the cache making start-up time faster.

*However*, if you are using local storage and set `ignoreCache` to true, it will ignore this setting when it actually comes to call the `/oauth/token` endpoint and read the refresh token from the cache regardless (otherwise this whole feature doesn't make any sense - the refresh token needs to be persisted).

Given that, I'm happy to reverse this change.

Reply:
Up until the last paragraph, I agree with it. But then on the `/oauth/token` call if `ignoreCache` is true, why is it taking the refresh token from it anyway? Shouldn't it fail, given that the code was instructed to ignore the cache?

Reply:
> if you are using local storage and set ignoreCache to true

This isn't quite correct, apologies - it's actually "if you are using refresh tokens and set `ignoreCache` to `true`" it will still read the refresh token from the cache. Regardless if it's in-memory or local storage.

> why is it taking the refresh token from it anyway

Because in the cache is precisely where the refresh token is stored, and it *needs* that if you have configured the client to use refresh tokens. Otherwise, you will never be able to rehydrate your session on startup using a refresh token because we wouldn't be able to retrieve it from the cache.

It's not really bypassing the option to ignore the cache as such. The purpose of that flag is to prevent the client from making that call to the token endpoint to get a new access token if it doesn't need to. Or you can force it by setting `ignoreCache: true`. Here, we're still making the call to `/oauth/token`, it's just we need to read the cache to get the refresh token to send to that endpoint.

Does that make it any more clear? Very happy to have a call tomorrow about this.